packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -617,6 +628,23 @@ export class Cluster extends Resource implements ICluster {
     return this._awsAuth;
   }
 
+  /**
+   * Lazily creates the OpenIdConnectProvider resource, which links the cluster to AWS IAM.
+   */
+  public get openIdConnectProvider() {
+    if (!this.kubectlEnabled) {
+      throw new Error('Cannot specify a OpenID Connect Provider if kubectl is disabled');
+    }
+
+    if (!this._openIdConnectProvider) {
+      this._openIdConnectProvider = new iam.OpenIdConnectProvider(this, 'OpenIdConnectProvider', {
+        url: this._clusterOpenIdConnectIssuerUrl!,
```

Comment:
Now you can safely do this:

```suggestion
        url: this.clusterOpenIdConnectIssuerUrl,
```